src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/LPG2RDFConfigurationReader.java

```diff
@@ -0,0 +1,112 @@
+
+        if (!lpg2rdfConfigs.hasNext()){
+            throw new IllegalArgumentException("LPGtoRDFConfiguration is required!");
+        }
+        else {
+            final Resource lpg2rdfConfig = lpg2rdfConfigs.next();
+
+            verifyLPG2RDFConfiguration(lpg2rdfConfig);
+
+            if(!lpg2rdfConfig.hasProperty(LPG2RDF.labelPredicate)){
+                throw new IllegalArgumentException("labelPredicate is required!");
+            }
+            final RDFNode labelPredicateObj = lpg2rdfConfig.getProperty(LPG2RDF.labelPredicate).getObject();
+            String url;
+            try {
+                url = labelPredicateObj.asResource().getURI();
+            }
+            catch (ResourceRequiredException re){
+                throw new IllegalArgumentException("labelPredicate is invalid!");
+            }
+            label = verifyExpectedURI(url).toString();
+
+            final Resource nodeMappingResource = lpg2rdfConfig.getProperty(LPG2RDF.nodeMapping).getResource();
+            final RDFNode nodeMappingResourceType = lpg2Rdf.getRequiredProperty(nodeMappingResource, RDF.type).getObject();
+
+            if ( nodeMappingResourceType.equals(LPG2RDF.IRIBasedNodeMapping) ) {
+                if(!nodeMappingResource.hasProperty(LPG2RDF.prefixOfIRIs)){
+                    throw new IllegalArgumentException("prefixOfIRIs is required!");
+                }
+                final RDFNode prefixOfIRIObj = nodeMappingResource.getProperty(LPG2RDF.prefixOfIRIs).getObject();
+                final String prefixOfIRIUrl = verifyExpectedURI(prefixOfIRIObj.asResource().getURI()).toString();
+                nodeMapping = new NodeMappingToURIsImpl(prefixOfIRIUrl);
+            }
+            else if ( nodeMappingResourceType.equals(LPG2RDF.BNodeBasedNodeMapping) ) {
+                nodeMapping = new NodeMappingToBNodesImpl();
+            }
+            else {
+                throw new IllegalArgumentException("Type of Resource in NodeMapping is not defined.");
+            }
+        }
+        return new LPG2RDFConfigurationImpl(NodeFactory.createURI(label), nodeMapping);
+    }
+
+
+    /**
+     * Verifies that the given string represents an HTTP URI
+     * or an HTTPS URI and, if so, returns that URI.
+     */
+    protected URI verifyExpectedURI(final String uriString ) {
+        final URI uri;
+        try {
+            uri = new URI(uriString);
+            if (    ! uri.isAbsolute()
+                    || (uri.getScheme().equals("http") && uri.getScheme().equals("https")) ) {
+                throw new IllegalArgumentException( "The following URI is of an unexpected type; it should be an HTTP URI or an HTTPS URI: " + uriString );
+            }
+        }
+        catch ( final URISyntaxException ex ) {
+            throw new IllegalArgumentException("URI parse exception (" + ex.getMessage() + ")");
+        }
+
+        return uri;
+    }
```

Comment:
As per my comments above, I don't think a helper function with this functionality is needed in this class.